### PR TITLE
Fix dynamic certificates issues

### PR DESCRIPTION
**Pull request by enj**
This change addresses three bugs:

1. Accidental mutex copy via TLS config struct copy
2. `nil` pointer exception caused by dereferencing the wrong variable
3. Pass in a valid related object to `Eventf` so that an event is actually created

Extracted out of #88682 to make the other PR easier to review.

/kind bug
/priority important-soon
/assign @deads2k 
/milestone v1.18

```release-note
NONE
```

### CODE REVIEW

**Comment by jackkleeman** [comment]
LGTM

File: staging/src/k8s.io/apiserver/pkg/server/dynamiccertificates/server_test.go [hunk 1]
```diff
@@ -79,7 +79,7 @@ func TestServingCert(t *testing.T) {
 	}
 
 	dynamicCertificateController := NewDynamicServingCertificateController(
-		*tlsConfig,
+		tlsConfig,
```
Comment (liggitt) [inline staging/src/k8s.io/apiserver/pkg/server/dynamiccertificates/server_test.go]: prefer `tlsConfig.Clone()` to make it obvious we're giving away a copy

File: staging/src/k8s.io/apiserver/pkg/server/secure_serving.go [hunk 2]
```diff
@@ -66,7 +66,7 @@ func (s *SecureServingInfo) tlsConfig(stopCh <-chan struct{}) (*tls.Config, erro
 
 	if s.ClientCA != nil || s.Cert != nil || len(s.SNICerts) > 0 {
 		dynamicCertificateController := dynamiccertificates.NewDynamicServingCertificateController(
-			*tlsConfig,
+			tlsConfig,
```
Comment (liggitt) [inline staging/src/k8s.io/apiserver/pkg/server/secure_serving.go]: prefer `tlsConfig.Clone()` to make it obvious we're giving away a copy

**Comment by liggitt** [comment]
one nit on cloning tls config before propagating... could also squash these to a single commit

**Pull request merged**